test/api_addProductFields_test.dart

```diff
@@ -0,0 +1,114 @@
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:openfoodfacts/utils/AddableProductFields.dart';
+import 'package:openfoodfacts/utils/OpenFoodAPIConfiguration.dart';
+import 'package:openfoodfacts/utils/QueryType.dart';
+import 'package:test/test.dart';
+import 'test_constants.dart';
+
+/// Unit/Integration tests around OpenFoodAPIClient.addProductFields
+void main() {
+  OpenFoodAPIConfiguration.globalQueryType = QueryType.TEST;
+
+  group('$OpenFoodAPIClient add product fields', () {
+    const String barcode = '0048151623426';
+    const List<String> additionalValues = <String>['djobi', 'djoba'];
+    const OpenFoodFactsLanguage language = OpenFoodFactsLanguage.ENGLISH;
+    const User user = TestConstants.TEST_USER;
+    const Timeout timeout = Timeout(Duration(seconds: 90));
+
+    void _checkStatus(final Status status) {
+      expect(status.status, 1);
+      expect(status.statusVerbose, 'fields saved');
+    }
+
+    String? _getValue(
+      final AddableProductField field,
+      final Product product,
+    ) {
+      switch (field) {
+        case AddableProductField.BRANDS:
+          return product.brands;
+        case AddableProductField.COUNTRIES:
+          return product.countries;
+        case AddableProductField.STORES:
+          return product.stores;
+      }
+    }
+
+    Future<void> _checkValue(
+      final AddableProductField field,
+      final String expectedValue,
+    ) async {
+      final ProductQueryConfiguration configuration = ProductQueryConfiguration(
+        barcode,
+        language: language,
+        fields: [ProductField.ALL],
+      );
+
+      final ProductResult productResult = await OpenFoodAPIClient.getProduct(
+        configuration,
+        user: user,
+      );
+      expect(productResult.product, isNotNull);
+      final String? actualValue = _getValue(field, productResult.product!);
+      expect(actualValue, equalsIgnoringCase(expectedValue));
+    }
+
+    Future<void> _checkAddableField(
+      final AddableProductField field,
+      final Product initialProduct,
+    ) async {
+      late Status status;
+      late Product product;
+
+      // from scratch
+      product = initialProduct;
+      status = await OpenFoodAPIClient.saveProduct(user, product);
+      _checkStatus(status);
+
+      // cumulative list
+      String expectedValue = _getValue(field, product)!;
```

Comment:
A `StringBuffer` would be better, since you concatenate multiple times

Reply:
@g123k `StringBuffer` is never used in the whole project. Here we're talking about adding a String to a String twice, and checking the resulting value 3 times (that would mean 3 additional `toString()` calls with `StringBuffer`). In a test. That makes async calls to the back-end that take longer time than a String allocation.

Reply:
@g123k I had a hunch about `StringBuffer` not being significantly faster than `String` when you only concatenate two `String`s and convert each time the result as a `String`.
I've coded a test with the 2 solutions: with `String` and with `StringBuffer`.
<details>
<summary>test code</summary>

```dart
      const String initialString = "init";
      const List<String> additionalValues = <String>['djobi', 'djoba'];
      final List<String> result = <String>[];

      late int start;
      late int end;

      result.clear();
      start = DateTime.now().microsecondsSinceEpoch;
      final StringBuffer asStringBuffer = StringBuffer(initialString);
      result.add(asStringBuffer.toString());
      for (final String additionalValue in additionalValues) {
        asStringBuffer.write(', $additionalValue');
        result.add(asStringBuffer.toString());
      }
      end = DateTime.now().microsecondsSinceEpoch;
      print('${end - start}: $result');

      result.clear();
      start = DateTime.now().microsecondsSinceEpoch;
      String asString = initialString;
      result.add(asString);
      for (final String additionalValue in additionalValues) {
        asString += ', $additionalValue';
        result.add(asString);
      }
      end = DateTime.now().microsecondsSinceEpoch;
      print('${end - start}: $result');
```
</details>

After several runs, I've noticed that the first block run is a bit slower than the other - regardless of `StringBuffer` and `String`. By approximately 1ms.
Can you afford that 1ms uncertainty for _tests_? I think so. If that's OK with you, please approve my PR. If you think there is a hidden performance potential, don't hesitate to keep this PR rejected and to run performance tests.